autoPyTorch/pipeline/components/training/trainer/RowCutOutTrainer.py

```diff
@@ -34,17 +36,26 @@ def data_preparation(self, X: np.ndarray, y: np.ndarray,
             lam = 1
             return X, {'y_a': y_a, 'y_b': y_b, 'lam': lam}
 
-        # The mixup component mixes up also on the batch dimension
-        # It is unlikely that the batch size is lower than the number of features, but
-        # be safe
-        size = min(X.shape[0], X.shape[1])
-        indices = torch.tensor(self.random_state.choice(range(1, size), max(1, np.int(size * self.patch_ratio))))
+        size = X.shape[1]
+        indices = self.random_state.choice(range(1, size), max(1, np.int32(size * self.patch_ratio)),
+                                           replace=False)
 
-        # We use an ordinal encoder on the tabular data
+        if not isinstance(self.numerical_columns, typing.Iterable):
```

Comment:
What if the numerical columns are None, we should still continue with only categorical imputing in this case or not.

Reply:
Also if there are only numerical columns, there should not be a conversion for categorical ones.

Reply:
Actually when there are no numerical columns, it is not none but it is an empty list. And indexing with an empty list does not affect the tensor so this should work

Reply:
https://github.com/automl/Auto-PyTorch/blob/8b71ee2ad3c00277928ac462c0d364151d69aca0/autoPyTorch/pipeline/components/training/trainer/base_trainer_choice.py#L340-L341

Is 'numerical_columns' always in 'dataset_properties' ?

Reply:
when its tabular data then yeah